### PR TITLE
Schema update: Adding data element definitions

### PR DESCRIPTION
**Pull request by talkasab**
Proposing adding more detail about the definition of data elements into the definition of a data element set.

Included a sample XML file which validates that demonstrates what this might look like for a subset of the adrenal nodule data element set.

Also tweaked the schema definition to be more specific about what's expected in the different attribute types.

### CODE REVIEW

File: CDE_schema.rnc [hunk 1]
```diff
@@ -4,13 +4,14 @@ start =
   element data_element_set {
     
     ## Each set has a unique identifier of the format "RDES[0-9]+" (e.g., "RDES42") and a textual name.
-    attribute id { text },
-    attribute name { text },
-    (attribute description { text }
-     & attribute url { xsd:anyURI }
+    attribute id { xsd:string { pattern='RDES\d+' } },
+    attribute shortName { xsd:Name }?,
```
Comment (shadowdoc) [inline CDE_schema.rnc:8]: What's the rationale for including a shortName in addition to Name?

File: CDE_schema.rnc [hunk 2]
```diff
@@ -73,10 +102,12 @@ image =
     & element source { text }?
     & authors?
   }
-history =
-  element event {
+status_attrs =
```
Comment (shadowdoc) [inline CDE_schema.rnc:105]: I think it's a good idea to generalize status at several levels, but I don't think it should be named status_attrs.   status works just fine.

File: CDE_schema.rnc [hunk 3]
```diff
@@ -57,11 +86,11 @@ org =
...
     element index_code {
-      attribute scheme { text },
-      attribute code { text },
-      attribute display_name { text }?,
-      attribute url { xsd:anyURI }?
-    }*
+      attribute system { xsd:Name },
+      attribute code { xsd:normalizedString },
+      attribute href { xsd:anyURI }?,
+      attribute display { text }?
```
Comment (shadowdoc) [inline CDE_schema.rnc:92]: If we're mirroring FHIR - https://www.hl7.org/fhir/datatypes.html#Coding
`
attribute system { xsd:anyURI },
attribute code { text },
attribute display { text },
`

They don't include HREF, and I don't think we need to either.  They do include version - which I'm +/- on including.

File: CDE_schema.rnc [hunk 4]
```diff
@@ -73,10 +102,12 @@ image =
...
     & authors?
   }
-history =
-  element event {
+status_attrs = 
     attribute date { xsd:date },
     attribute status {
       "proposed" | "approved" | "published" | "deprecated"
     }
+history =
```
Comment (shadowdoc) [inline CDE_schema.rnc:110]: I'm not sure that we need to nest stautus inside of two other tags

**Review by shadowdoc — DISMISSED** [review]
Thanks Terry - this is a huge amount of work!!  Please feel free to continue the dialog in the pull request.

File: CDE_schema.rnc (hunk 2 above)
Comment (talkasab) [reply CDE_schema.rnc:105]: "status_attrs" is just an internal name for the schema definition so we don't have to re-type this set of attributes wherever we use them.

File: CDE_schema.rnc (hunk 1 above)
Comment (talkasab) [reply CDE_schema.rnc:8]: Partly it's a use case thing; we use identifier-style names for the data elements internally within CAR/DS modules, and it makes working with them lots easier (than, for example, having to remember the standard IDs).
But we can skip that at the CDE level, I suppose.

File: CDE_schema.rnc (hunk 3 above)
Comment (talkasab) [reply CDE_schema.rnc:92]: I'm also meh on including the version.
I think allowing inclusion of the href (it's optional) would be useful, though.

File: CDE_schema.rnc (hunk 4 above)
Comment (talkasab) [reply CDE_schema.rnc:110]: "Status" isn't nested in multiple tags... it's directly on the "version" element (see the sample XML for examples).

**Comment by talkasab** [comment]
I'll work on some of the proposed changes in the next few days, but also see comments above.

File: CDE_schema.rnc (hunk 1 above)
Comment (shadowdoc) [reply CDE_schema.rnc:8]: Thanks - I think it's probably best to try to keep the schema simple as we know that complexity will grow over time.  

If you could submit another commit that removes the shortName attribute, that would be awesome.

File: CDE_schema.rnc (hunk 2 above)
Comment (shadowdoc) [reply CDE_schema.rnc:105]: Ok gotcha

File: CDE_schema.rnc (hunk 4 above)
Comment (shadowdoc) [reply CDE_schema.rnc:110]: Understood, sorry for the confusion.

File: CDE_schema.rnc (hunk 3 above)
Comment (shadowdoc) [reply CDE_schema.rnc:92]: Ok:

Let's leave out version.
Let's update system to be a URI
and include HREF as an optional element

Can you submit another commit?

**Comment by shadowdoc** [comment]
Hey @talkasab - we're down to just two issues that need a couple of commits.  Should be ~10 mins of work.  Let's put a nail in this one!

**Pull request merged**